src/server.cpp

```diff
@@ -318,6 +321,67 @@ int Server<IoType, SwitchActivityInfo, SwitchCalcGaps>::server_accept(int ifd) {
                                         inet_ntoa(addr.sin_addr), ntohs(addr.sin_port), active_ifd);
                             }
                             do_accept = true;
+
+                            if (s_user_params.sockperfDumpDataToFile)
+                            {
+                                retVal = system((createSockperfDirectoryCommand.append(s_user_params.sockperfDumpDataDirectory)).c_str());
+                                if(retVal != 0)
+                                {
+                                    log_msg("%s","unable to create directory structure");
+                                }
+                                sockperfDataFile = generateDumpFileName(inet_ntoa(addr.sin_addr));
+								
+								int fd = 0;
+								
+								switch (s_user_params.syncFlag)
+								{
+									case 1:
+									    log_msg("Got the request under case 1. Opening file");
+									    fd = open(sockperfDataFile.c_str(), O_CREAT | O_RDWR | O_DSYNC, "w+");
```

Comment:
Is "w+" needed here? since it is already included in fdopen.

Reply:
Yes, as the outer policy (fdopen) should be less restrictive.